pulsar-broker/src/main/java/org/apache/pulsar/compaction/StrategicTwoPhaseCompactor.java

```diff
@@ -377,7 +356,7 @@ private <T> CompletableFuture<Long> runPhaseTwo(
                                     return;
                                 }
                             });
-                })
+                }, scheduler)
```

Comment:
Is there any case when `loopPromise` is not completed in the `scheduler`? In `phaseTwoLoop`:

```java
        CompletableFuture.runAsync(() -> {
                    if (reader.hasNext()) {
                        /* ... */
                    } else {
                        promise.complete(null);
                        return;
                    }
                }, scheduler)
```

`promise.complete(null);` is called in the `scheduler`, I think we can immediately execute the callback of `thenCompose` instead adding the callback again to `scheduler`.

Reply:
Oh, I changed back. This change is made when I am debugging, this change is no useful.